framework/src/controller/schema/application.js

```diff
@@ -6,8 +6,144 @@ module.exports = {
 	},
 
 	genesisBlock: {
-		id: '#genesisBlock',
+		$id: '#genesisBlock',
 		type: 'object',
+		required: [
+			'version',
+			'totalAmount',
+			'totalFee',
+			'reward',
+			'payloadHash',
+			'timestamp',
+			'numberOfTransactions',
+			'payloadLength',
+			'generatorPublicKey',
+			'transactions',
+			'blockSignature',
+		],
+		properties: {
+			version: {
+				type: 'integer',
+				minimum: 0,
+			},
+			totalAmount: {
+				type: 'string',
+				format: 'amount',
+			},
+			totalFee: {
+				type: 'number',
+				format: 'maxAmount',
+			},
+			reward: {
+				type: 'number',
+				format: 'maxAmount',
+			},
+			payloadHash: {
+				type: 'string',
+				format: 'hex',
+			},
+			timestamp: {
+				type: 'integer',
+				min: 0,
+			},
+			numberOfTransactions: {
+				type: 'integer',
+				min: 0,
+			},
+			payloadLength: {
+				type: 'integer',
+				min: 0,
+			},
+			previousBlock: {
+				type: ['null', 'string'],
+				format: 'id',
+				minLength: 1,
+				maxLength: 20,
+			},
+			generatorPublicKey: {
+				type: 'string',
+				format: 'publicKey',
+			},
+			transactions: {
+				type: 'array',
+				items: {
+					$ref: 'transactions',
+				},
+				uniqueItems: true,
+			},
+			height: {
+				type: 'integer',
+				min: 1,
+			},
+			blockSignature: {
+				type: 'string',
+				format: 'signature',
+			},
+			id: {
+				type: 'string',
+				format: 'id',
+				minLength: 1,
+				maxLength: 20,
+			},
+		},
+		additionalProperties: false,
+	},
+
+	transactions: {
+		id: 'transactions',
+		type: 'object',
+		required: ['type', 'timestamp', 'senderPublicKey', 'signature'],
+		properties: {
+			type: {
+				type: 'integer',
+				minimum: 0,
+				maximum: 7,
```

Comment:
In genesis block we can't have all transactions.

Reply:
Pushed new commit: enum that allows only type `0`, `2`, and `3`. (The types that occur in the `genesis_block.json`.